src/includes/product.ts

```diff
@@ -16,6 +16,14 @@ export interface Product extends Record<string, any> {
   variants?: ProductVariants;
 }
 
+/**
+ * The type of a function used to transform record data into products.
+ *
+ * @typeparam T The type of the product into which the Record will be
+ * transformed.
+*/
+export type ProductTransformer<T> = (product: Record<string, any>) => T;
```

Comment:
You should be able to add documentation for the `product` param and the return value as well here with the usual method (`@param` and `@returns`).

If not, you can consider turning it into an interface:

```typescript
/**
 * ...
 *
 * @typeparam T ...
 */
export interface ProductTransformer<T> {
  /**
   * ...
   *
   * @param product ...
   */
  (product: Record<string, any>): T;
}
```